tests/e2e-tests/specs/wp-admin/wp-admin-settings-tax.test.js

```diff
@@ -0,0 +1,137 @@
+/**
+ * @format
+ */
+
+/**
+ * External dependencies
+ */
+import { activatePlugin } from '@wordpress/e2e-test-utils';
+
+/**
+ * Internal dependencies
+ */
+import { StoreOwnerFlow } from '../../utils/flows';
+import { clearAndFillInput, setCheckbox, settingsPageSaveChanges, uiUnblocked } from "../../utils";
+
+describe( 'WooCommerce Tax Settings', () => {
+	beforeAll( async () => {
+		await activatePlugin( 'woocommerce' );
+	} );
+
+	it( 'can enable tax calculation', async() => {
+		// Go to general settings page
+		await StoreOwnerFlow.openSettings( 'general' );
+
+		// Make sure the general tab is active
+		await expect( page ).toMatchElement( 'a.nav-tab-active', { text: 'General' } );
+
+		// Enable tax calculation
+		await setCheckbox( 'input[name="woocommerce_calc_taxes"]' );
+		await settingsPageSaveChanges();
+		await expect( page ).toMatchElement( '#message', { text: 'Your settings have been saved.' } );
+
+		// Verify that tax settings are now present
+		await expect( page ).toMatchElement( 'a.nav-tab', { text: 'Tax' } );
+	} );
+
+	it( 'can set tax options', async () => {
+		// Go to tax settings page
+		await StoreOwnerFlow.openSettings( 'tax' );
+
+		// Make sure the tax tab is active
+		await expect( page ).toMatchElement( 'a.nav-tab-active', { text: 'Tax' } );
+
+		// Prices exclusive of tax
+		await expect( page ).toClick( 'input[name="woocommerce_prices_include_tax"][value="no"]' );
+		// Tax based on customer shipping address
+		await expect( page ).toSelect( '#woocommerce_tax_based_on', 'Customer shipping address' );
+		// Standard tax class for shipping
+		await expect( page ).toSelect( '#woocommerce_shipping_tax_class', 'Standard' );
+		// Leave rounding unchecked (no-op)
+		// Display prices excluding tax
+		await expect( page ).toSelect( '#woocommerce_tax_display_shop', 'Excluding tax' );
+		// Display prices including tax in cart and at checkout
+		await expect( page ).toSelect( '#woocommerce_tax_display_cart', 'Including tax' );
+		// Display a single tax total
+		await expect( page ).toSelect( '#woocommerce_tax_total_display', 'As a single total' );
+
+		await settingsPageSaveChanges();
+		await expect( page ).toMatchElement( '#message', { text: 'Your settings have been saved.' } );
```

Comment:
As per the previous settings PR, I think we should actually test that the settings are keeping state after the page reloads instead of just relying on the `Your settings have been saved` message.

Reply:
> As per the previous settings PR, I think we should actually test that the settings are keeping state after the page reloads instead of just relying on the Your settings have been saved message.

@kloon I agree. Will work on the verification and will send another commit once done. Thanks for a thorough review.